lib/elixir/lib/code/formatter.ex

```diff
@@ -2450,4 +2473,47 @@ defmodule Code.Formatter do
   defp has_double_quote?(chunk) do
     is_binary(chunk) and chunk =~ @double_quote
   end
+
+  # Migration rewrites
+
+  @bool_operators [
+    :>,
+    :>=,
+    :<,
+    :<=,
+    :in
+  ]
+  @guards [
+    :is_atom,
+    :is_boolean,
+    :is_nil,
+    :is_number,
+    :is_integer,
+    :is_float,
+    :is_binary,
+    :is_map,
+    :is_struct,
+    :is_non_struct_map,
+    :is_exception,
+    :is_list,
+    :is_tuple,
+    :is_function,
+    :is_reference,
+    :is_pid,
+    :is_port
+  ]
+
+  defp negate_condition(condition) do
+    case condition do
+      {neg, _, [condition]} when neg in [:!, :not] -> condition
+      {:|>, _, _} -> {:|>, [], [condition, {{:., [], [Kernel, :!]}, [closing: []], []}]}
```

Comment:
we found the migrations more readable _not_ doing this particular transformation, as this hides the most impactful part of the statement. compare:

```elixir
# given:
unless Widget |> where(foo: ^bar) |> Repo.exists?(), do: ...
# rewritten to if with pipe
if Widget |> where(foo: ^bar) |> Repo.exists?() |> Kernel.!(), do: ...
# treated as any other statement
if !(Widget |> where(foo: ^bar) |> Repo.exists?()), do:
```
our team [asked for the latter](https://github.com/adobe/elixir-styler/commit/4aa18ba4fb7a2232bbe337fe43afff4e1387d67b) as it has a much more natural-language read (not to mention that `Kernel.!()` is unlikely to be written by a human in any case)

Reply:
Thank you for the suggestion 💜 Even if I'm the one who originally implemented it as `|> Kernel.!()`, now that you mention it I have to agree.

Plus this is removing complexity so I'm all for it.

Do you want to send a PR @novaugust?

Reply:
happy to, i'll kick one out tomorrow